File: 2971-find-polygon-with-the-largest-perimeter/2971-find-polygon-with-the-largest-perimeter.py
class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        summ=0
        nums.sort()
        for i in range(len(nums)):
            summ+=nums[i]
        
        for i in range(len(nums)-1,-1,-1):
            summ-=nums[i]
            if(summ>nums[i]):
                return summ+nums[i]
        return -1
    
 

File: 2971-find-polygon-with-the-largest-perimeter/2971-find-polygon-with-the-largest-perimeter.java

class Solution {
    public long largestPerimeter(int[] nums) {
        long sum=0;
        Arrays.sort(nums);
        for(int i=0;i<nums.length;i++){
            sum+=nums[i];
        }
        for(int i=nums.length-1;i>=0;i--){
            sum-=nums[i];
            if(sum>nums[i]){
                return sum+nums[i];
            }
        }
        return -1;
    }
}